Named and typed Extensions

retothuerer opened the pull request:
- Specification for type and name
- added railroad diagram
- sub page with well known types

Commit:
- added railroad diagram
- mention type and name
- sub page with well known types

retothuerer committed:
fixed incorrect name of type

retothuerer committed:
- introduction text for extension
- replaces png by svg image
- formatting of standing terms

retothuerer committed:
Merge branch 'improvements-to-documentation-of-extensions' of https://github.com/retothuerer/PAC-ID into improvements-to-documentation-of-extensions

apinilabs-manuel commented on README.md:
```diff
@@ -117,10 +117,25 @@ Using multiple `id segment`s might seem like unnecessary overhead at first. Addi
 - Using the key/value syntax (`id segment key` `:` `id segment value`) with [well-known `id segment keys`](well-known-id-segment-keys.md) enables `PAC-ID` routing in generic contexts.
 
 ## PAC-ID Extension
+`PAC-ID`s can be extended with custom information after a * character. The * character and subsequent characters are not part of the `PAC-ID` and can therefore freely be used for one or multiple `extension`s. In order to facilitate parsing and rendering by systems like [PAC-ID resolver](https://github.com/ApiniLabs/PAC-ID-Resolver), `name` and `type` of the `extension` MUST be specified.
```
I'd regard name and type of extensions as optional because overall length is a concern. Name and type are mainly useful when there are multiple extensions present.

apinilabs-daniel replied:
After discussion with Manu, I suggest the following:
Explicitly specify the structure of an extension, e.g. also explain that it is from * to * and that the * do not count to the extension:
- anything after the "*" until the end of text or the next "*" (not inlcuding) is an "extension"
- after the * a name/type$ CAN indicate name and type,  name and type are RECOMMENDED only --> if no name/type present, it is RECOMMENDED that the extension is a high level summary of what the PAC-ID points to in the T-REX format.
- `/` is NOT allowed as part of an extension (reserved to indicate name/type)
- introduce / mark terms properly with `` 

Remove the sentence "in order to facilitate parsing ... ... specified."

apinilabs-manuel commented on README.md:
````diff
@@ -117,10 +117,25 @@ Using multiple `id segment`s might seem like unnecessary overhead at first. Addi
+
+An example of a PAC-ID with two extensions, one describing the calibration state of a balance, the other it's current measurements:
+```
+HTTPS://PAC.METTORIUS.COM/-MD/240:BAL500/21:210263*CAL$TREX/START$T.D:20240822+PF$T.B:T+17$T.D:20250822*CUR$TREX/TARE$MGM:2.5E2+ENV$KEL:293.15
````
While this is obviously perfectly valid, I'd still recommend to always follow the same pattern of using the T-REX as a summary of the item referred to in the PAC for having coherent examples that would also scale to more complex scenarios where there would be actual datasets.

apinilabs-daniel replied:
I agree to Manu, I would make 2 examples, with and without name/type and use "SUM" as a name, indicating summary

one pointing to a device
- with HTTPS://PAC.METTORIUS.COM/-MD/240:BAL500/21:210263*SUM$TREX/START$T.D:20240822+PF$T.B:T+17$T.D:20250822 
- and without name HTTPS://PAC.METTORIUS.COM/-MD/240:BAL500/21:210263*START$T.D:20240822+PF$T.B:T+17$T.D:20250822

and one pointing to a result data ( HTTPS://PAC.METTORIUS.COM/-MR/... with and without name/type

apinilabs-manuel commented on well-known-extension-types.md:
```diff
@@ -0,0 +1,5 @@
+
+| Type | Description | Further information |
+|:--- |:--- |:--- |
+|TREX | T-REX (Trivial Record Exchange) is a simple text-based serialization format. It is purpose-built for replacing manual transcription scenarios with easy to implement ad-hoc data transmissions.| https://github.com/ApiniLabs/T-REX|
```
Might be worthwhile mentioning that T-REX is usually used for providing a generic summary of what the PAC-ID is pointing at.

apinilabs-daniel replied:
In addition, to save characters, I recommend that we use "T" or "TX" instead of "TREX" as type.

retothuerer committed:
typo

retothuerer committed:
incorporated suggestions from PR review
- name and type optional
- first extension should be a summary of what the PAC-ID points to

retothuerer committed:
Merge branch 'improvements-to-documentation-of-extensions' of https://github.com/retothuerer/PAC-ID into improvements-to-documentation-of-extensions

retothuerer committed:
fixed formatting error

retothuerer committed:
correction of duplicated text (merge mistake)

apinilabs-daniel committed:
Update README.md

removed duplicate line

apinilabs-daniel committed:
Update README.md

corrected some minor spelling inconsistencies.
